File: objectiveai-sdk-js/src/functions/executions/http.ts
import { ObjectiveAI, type RequestOptions } from "../../client";
import { Stream } from "../../stream";
import type { FunctionsExecutionsRequestFunctionExecutionCreateParams } from "./request/functionExecutionCreateParams";
import type { FunctionsExecutionsResponseUnaryFunctionExecution } from "./response/unary/functionExecution";
import type { FunctionsExecutionsResponseStreamingFunctionExecutionChunk } from "./response/streaming/functionExecutionChunk";

export function functionsExecutionsCreateFunctionExecution(
  client: ObjectiveAI,
  body: FunctionsExecutionsRequestFunctionExecutionCreateParams & { stream: true },
  options?: RequestOptions,
): Promise<Stream<FunctionsExecutionsResponseStreamingFunctionExecutionChunk>>;
export function functionsExecutionsCreateFunctionExecution(
  client: ObjectiveAI,
  body: FunctionsExecutionsRequestFunctionExecutionCreateParams & { stream?: false | null },
  options?: RequestOptions,
): Promise<FunctionsExecutionsResponseUnaryFunctionExecution>;
export function functionsExecutionsCreateFunctionExecution(
  client: ObjectiveAI,
  body: FunctionsExecutionsRequestFunctionExecutionCreateParams,
  options?: RequestOptions,
): Promise<
  | Stream<FunctionsExecutionsResponseStreamingFunctionExecutionChunk>
  | FunctionsExecutionsResponseUnaryFunctionExecution
> {
  if (body.stream) {
    return client.post_streaming<FunctionsExecutionsResponseStreamingFunctionExecutionChunk>(
      "functions/executions",
      body,
      options,
    );
  }
  return client.post_unary<FunctionsExecutionsResponseUnaryFunctionExecution>(
    "functions/executions",
    body,
    options,
  );
}

File: objectiveai-sdk-py/objectiveai_sdk/functions/executions/http.py
"""HTTP functions for function executions."""

from __future__ import annotations

from typing import TYPE_CHECKING, Union

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.functions.executions.request import FunctionExecutionCreateParams
    from objectiveai_sdk.functions.executions.response.streaming import FunctionExecutionChunk
    from objectiveai_sdk.functions.executions.response.unary import FunctionExecution
    from objectiveai_sdk.stream import Stream


async def create_function_execution(
    client: ObjectiveAI,
    params: FunctionExecutionCreateParams,
) -> Union[FunctionExecution, Stream[FunctionExecutionChunk]]:
    """Execute a function.

    If ``params.stream`` is true, returns a streaming response.
    """
    if getattr(params, "stream", None):
        return await client.post_streaming("functions/executions", params)
    return await client.post_unary("functions/executions", params)
